syntax = "proto3";

import "google/protobuf/empty.proto";

message TestMessage {
  string text = 1;
}

message Address {
  int32 ip = 1;
  int32 port = 2;
}

message SearchMessage {
  Address addr = 1; // Address of file requester
  string fileName = 2;
  string identifier = 3;
}

message SearchResponse {
  string addr = 1; // Address of node having the file
}

message File {
    string name = 1;
    string data = 2;
}

service PeerToPeer {
  rpc HeartBeat(google.protobuf.Empty) returns (google.protobuf.Empty) {}
  rpc TestRpc(TestMessage) returns (TestMessage) {}
  rpc SearchFile(SearchMessage) returns (google.protobuf.Empty) {}
}